nixos/modules/services/networking/fedimintd.nix

```
@@ -0,0 +1,363 @@
+            };
+
+            secretFile = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = ''
+                If set the URL specified in `bitcoin.rpc.url` will get the content of this file added
+                as an URL password, so `http://user@example.com` will turn into `http://user:SOMESECRET@example.com`.
+
+                Example:
+
+                `/etc/nix-bitcoin-secrets/bitcoin-rpcpassword-public` (for nix-bitcoin default)
+              '';
+            };
+          };
+        };
+
+        consensus.finalityDelay = mkOption {
+          type = types.number;
+          default = 10;
+          description = "Consensus peg-in finality delay.";
+        };
+
+        dataDir = mkOption {
+          type = types.str;
+          default = "/var/lib/fedimintd-${name}/";
+          readOnly = true;
+          description = ''
+            Path to the data dir fedimintd will use to store its data.
+            Note that due to using the DynamicUser feature of systemd, this value should not be changed
+            and is set to be read only.
+          '';
+        };
+
+        nginx = {
+          enable = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              Wether to configure nginx for fedimintd
+            '';
+          };
+          config = mkOption {
+            type = types.submodule (
+              recursiveUpdate (import ../web-servers/nginx/vhost-options.nix { inherit config lib; }) {
+                options.serverName = {
+                  default = "fedimintd-${name}";
+                  defaultText = "fedimintd-\${name}";
+                };
+              }
+            );
+            default = { };
+            description = "Overrides to the nginx vhost section for api";
+          };
+        };
+      };
+    };
+in
+{
+  options = {
+    services.fedimintd = mkOption {
+      type = types.attrsOf (types.submodule fedimintdOpts);
+      default = { };
+      description = "Specification of one or more fedimintd instances.";
+    };
+  };
+
+  config = mkIf (eachFedimintd != { }) {
+
+    assertions = flatten (
+      mapAttrsToList (fedimintdName: cfg: [
+        {
+          assertion = cfg.p2p.url != null;
+          message = ''
+            `services.fedimintd.${fedimintdName}.p2p.url` must be set to address reachable by other peers.
+
+            Example: `fedimint://p2p.mymint.org`.
+          '';
+        }
+        {
+          assertion = cfg.api.url != null;
+          message = ''
+            `services.fedimintd.${fedimintdName}.api.url` must be set to address reachable by the clients, with TLS terminated by external service (typically nginx), and relayed to the fedimintd bind address.
+
+            Example: `wss://api.mymint.org`.
+          '';
+        }
+      ]) eachFedimintd
+    );
+
+    networking.firewall.allowedTCPPorts = flatten (
```

Slightly more efficient since you know the list nesting only is one layer deep. Same for `assertions` maybe.

```suggestion
    networking.firewall.allowedTCPPorts = concatLists (
```